tooling/lsp/src/requests/mod.rs

```diff
@@ -327,6 +332,19 @@ fn on_formatting_inner(
     }
 }
 
+fn read_config(file_path: Option<&Path>) -> Config {
+    match file_path {
+        Some(file_path) => match Config::read(file_path) {
+            Ok(config) => config,
+            Err(err) => {
+                eprintln!("Failed to parse noirfmt.toml: {}", err);
```

Comment:
I'd print the full path, otherwise if there are multiple the user wouldn't know which one failed exactly.

Reply:
Ah never mind, that is part of the error 👍

Reply:
Ah, I thought about it but we don't always get the path in the error 😅 I'll change it to include it.

Reply:
Done! I change it to just print `err`, which now always includes the full path.

Reply:
Okay! A consequence here is that if there is an error in a deeply nested config, it reverts all of it to defaults, even if there is a non-default in one of the ancestors.

Reply:
I guess we could try again starting from the grandparent directory of the file that errored... but maybe for now this is good enough, if there's an error it might be noticeable and the user would fix it before continuing formatting everything.